### PR TITLE
Angular controller subgenerator and test added

### PR DESCRIPTION
**Pull request by UndeadBaneGitHub**

### CODE REVIEW

**Comment by codydaig** [comment]
The spacing in 2 of the 4 files needs to be changed to 2 spaces.

**Comment by UndeadBaneGitHub** [comment]
@codydaig Sorry, the templates were taken from old version of generators, fixing

**Comment by codydaig** [comment]
@UndeadBaneGitHub No worries! I appreciate the hard work! I'm going through all the PRs now and leaving comments if I have any. All PRs should be squashed too 1 commit. (Please squash and not just open a new one. It's super easy and I can walk you through it if needed.) Then they all just needs final review from @ilanbiala

File: angular-controller/templates/_.client.controller.tests.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    // Then we can start by loading the main application module
+    beforeEach(module(ApplicationConfiguration.applicationModuleName));
+
+    // The injector ignores leading and trailing underscores here (i.e. _$httpBackend_).
+    // This allows us to inject a service but then attach it to a variable
+    // with the same name as the service.
+    beforeEach(inject(function ($controller, $rootScope, _$location_, _$stateParams_, _$httpBackend_) {
+      // Set a new global scope
+      scope = $rootScope.$new();
```
Comment (ilanbiala) [inline angular-controller/templates/_.client.controller.tests.js]: $

File: angular-controller/templates/_.client.controller.tests.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+      // Point global variables to injected services
+      $stateParams = _$stateParams_;
+      $httpBackend = _$httpBackend_;
+      $location = _$location_;
+
+      // Initialize the <%= humanizedControllerName %> controller.
+      <%= classifiedControllerName %>Controller = $controller('<%= classifiedControllerName %>Controller', {
+        $scope: scope
```
Comment (ilanbiala) [inline angular-controller/templates/_.client.controller.tests.js]: $

File: test/angular-controller.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+var path = require('path'),
+  helpers = require('yeoman-generator').test,
+  assert = require('yeoman-generator').assert,
+  temp = require('temp').track();
+
+describe('AngularJS Sub Generators Tests', function () {
+  this.timeout(0);
```
Comment (ilanbiala) [inline test/angular-controller.test.js]: ?

**Comment by ilanbiala** [comment]
Mostly looks good, but I'm not understanding why we want no timeout and what the foo argument does?

File: angular-controller/templates/_.client.controller.tests.js (hunk 2 above)
Comment (UndeadBaneGitHub) [inline angular-controller/templates/_.client.controller.tests.js]: @ilanbiala Can you explain the comment please? Otherwise, pushing prompt version now.

Comment (ilanbiala) [inline angular-controller/templates/_.client.controller.tests.js]: scope -> $scope.

**Comment by codydaig** [comment]
@ilanbiala LGTY?

**Comment by ilanbiala** [comment]
LGTM.

**Pull request merged**